crates/frontend/src/components/monaco_editor.rs

```diff
@@ -1,20 +1,24 @@
+use std::borrow::Borrow;
 use std::{cell::RefCell, rc::Rc};
 
 use leptos::*;
 use monaco::api::CodeEditor;
 use monaco::sys::editor::{IEditorMinimapOptions, IStandaloneEditorConstructionOptions};
+use web_sys::KeyboardEvent;
 #[derive(Debug, Clone, strum_macros::Display)]
 #[strum(serialize_all = "lowercase")]
 pub enum Languages {
     Javascript,
     Json,
 }
 
-pub type EditorModelCell = Rc<RefCell<Option<CodeEditor>>>;
+pub type EditorModelCell = Rc<Option<CodeEditor>>;
 
 #[component]
 pub fn monaco_editor(
-    node_id: &'static str,
+    #[prop(into)] node_id: String,
+    #[prop(into, default = String::new())] data: String,
```

Comment:
Why take data separately?

Reply:
I want to remove the signals and use this `data` parameter, and handle any changes with `on_change` callback